home/utils.py

```diff
@@ -600,26 +655,63 @@ def from_platform_body_element(
         return message
 
 
+@dataclass
+class VariationMessage:
+    body: str = ""
+    title: str = ""
+
+    @classmethod
+    def from_variation(cls, variation: blocks.StreamValue.StreamChild):
+        message = cls()
+        message.body = variation.get("message")
+        profile_fields = []
+        for profile_field in variation.get("variation_restrictions").raw_data:
+            profile_fields.append(f"{profile_field['type']}: {profile_field['value']}")
+        message.title = ", ".join(profile_fields)
+        return message
+
+
+@dataclass
+class VariationMessageList:
+    variations: Tuple[VariationMessage]
+
+    @classmethod
+    def from_platform_body_element(cls, whatsapp_msg: blocks.StreamValue.StreamChild):
+        variations = []
+        for variation in whatsapp_msg.value["variation_messages"]:
+            variations.append(VariationMessage.from_variation(variation=variation))
+        return cls(variations)
+
+    def __iter__(self):
+        for value in self.__dict__.values():
+            yield value
```

Comment:
I'm unsure where we're using this `__iter__`? It's a bit of a strange way this is implemented, since the only value this class has is `variations`, this will just yield a single item, the value stored at variations. So why not just `yield self.variations`? Also, it might make more sense to do a `for message in self.variations: yield message`, so that we can do a `for variation in variation_message_list`, instead of a
```
for variations in variation_message_list:
  for variation in variations:
     ...
```